### PR TITLE
#316 remove tqdm

### PR DESCRIPTION
**Pull request by wuhu**
Removing tqdm.

#316

Output of `_pd_itercall` is now a `_GeneratorWithLength` instance. This is to give it a length such that when calling

```python
for x in tqdm(t.eval_apply(...)):
    ...
```

we get a progress bar instead of just a counter.

### CODE REVIEW

File: padl/transforms.py [hunk 1]
```diff
@@ -713,39 +725,29 @@ def _pd_itercall(self, args, mode: Mode, loader_kwargs: Optional[dict] = None,
...
-                    output_format = self._pd_get_output_format()
-                    if output_format is not None:
-                        yield output_format(*out)
-                    else:
-                        yield out
-            else:
-                output_format = self._pd_get_output_format()
-                if output_format is not None:
-                    yield output_format(*output)
+        def _gen():
```
Comment (sjrl) [inline padl/transforms.py:728]: Thanks for simplifying the logic here!

**Comment by jasonkhadka** [comment]
great! 
Should an example also be included in documentation to show how `tqdm` should be used along with `train_apply` and `eval_apply`?

**Comment by sjrl** [comment]
> 

I think that would be a great idea! Also we can update the notebook examples to use `tqdm` to show that as well. Speaking of which @wuhu can you check that the `verbose` option was not used in the notebooks and that it does not appear anywhere else in the docs?

**Comment by wuhu** [comment]
@jasonkhadka @sjrl thanks for the reviews! I've now double checked that we don't use verbose anywhere and added a section to the docs showing how to use it with tqdm.

**Review by jasonkhadka — APPROVED** [review]

**Review by sjrl — APPROVED** [review]

**Pull request merged**